### PR TITLE
[key-manager] Build initial rotation capability

### PR DESCRIPTION
**Pull request by davidiw**
This is really just to prove the end-to-end correctness of this tool.
After this everything becomes harder as the tool needs to be hardened
for fault tolerance.
Also found that the commit and execute needs to handle reconfigurations :O
@zekun000 ?

### CODE REVIEW

File: secure/key-manager/src/tests.rs [hunk 1]
```diff
@@ -37,13 +37,32 @@ struct Node {
...
+
+    sec_storage
+        .create(crate::CONSENSUS_KEY, c_prikey0, &Policy::public())
+        .unwrap();
+    // Ugly hack but we need this until we support retrieving a policy from within storage and that
+    // currently is not easy, since we would need to convert from Vault -> Libra policy.
     sec_storage
-        .create(crate::CONSENSUS_KEY, prikey, &Policy::public())
+        .create(
+            &format!("{}_previous", crate::CONSENSUS_KEY),
```
Comment (zekun000) [inline secure/key-manager/src/tests.rs]: which part is ugly hack? filling _previous? Policy::public?

File: secure/key-manager/src/tests.rs [hunk 2]
```diff
@@ -244,10 +281,10 @@ fn test_consensus_rotation() {
 }
 
 #[test]
-// This verifies the key manager is properly setup
-fn test_key_manager_init() {
+// This verifies the key manager is properly setup and that a
```
Comment (zekun000) [inline secure/key-manager/src/tests.rs]: not finished?

File: secure/key-manager/src/tests.rs [hunk 3]
```diff
@@ -79,6 +99,7 @@ fn setup(config: &NodeConfig) -> Node {
 
 #[derive(Clone)]
 struct TestLibraInterface {
+    queued_transactions: Arc<RefCell<Vec<Transaction>>>,
```
Comment (zekun000) [inline secure/key-manager/src/tests.rs]: why arc? I thought RefCell is enough?

File: secure/key-manager/src/tests.rs [hunk 4]
```diff
@@ -163,10 +197,13 @@ impl LibraInterface for TestLibraInterface {
     }
 }
 
+// TODO(davidiw): Notify external upon reconfigure and update epoch via ledger_info and also add
+// next validator set if appropriate...
```
Comment (zekun000) [inline secure/key-manager/src/tests.rs]: do we need this?

File: secure/key-manager/src/tests.rs (hunk 1 above)
Comment (davidiw) [reply secure/key-manager/src/tests.rs]: Yes, ideally this should be handled behind the scenes when generating the initial public key... creating the proper versioning at that point in time... or at least have a default.

File: secure/key-manager/src/tests.rs (hunk 3 above)
Comment (davidiw) [reply secure/key-manager/src/tests.rs]: We clone it for use within node and within key manager. I thought RefCell would be enough too until the test failed :D.

File: secure/key-manager/src/tests.rs (hunk 2 above)
Comment (davidiw) [reply secure/key-manager/src/tests.rs]: well the code is still incomplete too :P

File: secure/key-manager/src/tests.rs (hunk 4 above)
Comment (davidiw) [reply secure/key-manager/src/tests.rs]: Well the executor isn't really functionally complete...

**Pull request closed without merging**